include/highfive/bits/H5Easy_Eigen.hpp

```diff
@@ -0,0 +1,190 @@
+/*
+ *  Copyright (c), 2017, Adrien Devresse <adrien.devresse@epfl.ch>
+ *
+ *  Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ *
+ */
+#ifndef H5EASY_BITS_EIGEN_HPP
+#define H5EASY_BITS_EIGEN_HPP
+
+#include "../H5Easy.hpp"
+#include "H5Easy_misc.hpp"
+
+#ifdef HIGHFIVE_EIGEN
+
+namespace HighFive
+{
+    namespace detail
+    {
+        namespace eigen
+        {
+            template <class C, int Rows, int Cols, int Options>
+            std::vector<std::size_t> shape(const Eigen::Matrix<C,Rows,Cols,Options>& data)
+            {
+                if (Rows==1)
+                {
+                    return {static_cast<std::size_t>(data.cols())};
+                }
+
+                if (Cols==1)
+                {
+                    return {static_cast<std::size_t>(data.rows())};
+                }
+
+                return {static_cast<std::size_t>(data.rows()), static_cast<std::size_t>(data.cols())};
+            }
+
+            template <class C, int Rows, int Cols, int Options>
+            void write(HighFive::DataSet& dataset, const Eigen::Matrix<C,Rows,Cols,Options>& data)
+            {
+                if (data.IsRowMajor)
+                {
+                    dataset.write(data.data());
+                }
+                else
+                {
+                    Eigen::Matrix<C,Rows,Cols,Eigen::RowMajor> tmp = data;
```

Comment:
Is a copy required here?

Reply:
Maybe the `Map` functionality of Eigen can provide a way out here.

Reply:
@tristan0x Can you tell me if there is a way to write using to iterator (that loops over non-contiguous data)?